src/Features/CSharp/Portable/CodeCleanup/CSharpCodeCleanupService.cs

```diff
@@ -192,55 +185,42 @@ internal class CSharpCodeCleanupService : ICodeCleanupService
             return document;
         }
 
-        private async Task<Document> ApplyCodeFixesForSpecificDiagnosticIds(
-            Document document, ImmutableArray<string> diagnosticIds, CancellationToken cancellationToken)
+        private async Task<Document> ApplyCodeFixesForSpecificDiagnosticIdsAsync(
+            Document document, ImmutableArray<string> diagnosticIds, IProgressTracker progressTracker, CancellationToken cancellationToken)
         {
             foreach (var diagnosticId in diagnosticIds)
```

Comment:
we can get very fine-grained progress if we add all the diagnostic ids to the progress tracker.  Note: i would do this up front in "ApplyCodeFixesAsync" so that we have hte full amount of progress items known, instead of  increasing the amount on each call to this method.